### PR TITLE
Feature/event rewards system

### PR DESCRIPTION
**Pull request by tzwang1**

### CODE REVIEW

**Comment by tzwang1** [comment]
Things that still need to be done:
1. Decide on item rewards
2. Decide on how much each reward should cost
3. Decide on how many event points you should get for the number of correct answers
4. Something weird happened with it converting whitespace from the mac/linux format to the windows format. Should probably fix that also.

**Comment by tzwang1** [comment]
Current item rewards and their cost set to:
Chaos Scrolls (3 golden maple leaves)
Innocence Scrolls (7 golden maple leaves)
Onyx Apples (10 golden maple leaves)
White Scrolls (50 golden maple leaves)

Ox Quiz point rewards:
20 correct - 10 leaves
20 > correct >= 17 - 5 leaves
17 > correct >= 13 - 3 leaves
13 > correct >= 10 - 2 leaves
10 > correct - 1 leaf

File: scripts/npc/9000002.js [hunk 1]
```diff
@@ -55,6 +56,27 @@ function actionWithReward(mode, type, selection) {
     }
 }
 
+function getRewardThenWarp(cm, location) {
+    if (cm.canHold(golden_maple_leaf)) {
```
Comment (nansonzheng) [inline scripts/npc/9000002.js]: You should check for the amount of to hold as well in case of overflow

File: src/client/MapleCharacter.java [hunk 2]
```diff
@@ -7450,4 +7451,12 @@ public void resetBossEntries(){
         }
     }
 
+    public void setLatestEventPlacement(int latestEventPlacement) {
```
Comment (nansonzheng) [inline src/client/MapleCharacter.java:7454]: This won't get saved when the client logs out of a character, is this what you intended to do?

**Review by nansonzheng — COMMENTED** [review]
Found 2 possible sources of error

File: src/client/MapleCharacter.java (hunk 2 above)
Comment (tzwang1) [reply src/client/MapleCharacter.java:7454]: Nope it's not. I want this to be on the server so if someone crashes they can still get their event rewards. Their event placement should only be reset once they've collected their reward. I guess I would need to implement a handler for this then?

**Review by BenjixD — APPROVED** [review]

**Pull request merged**